app/bundles/CoreBundle/Tests/Unit/Command/src/FakeModeratedCommand.php

```diff
@@ -26,4 +27,17 @@ public function forceCompleteRun(): void
     {
         $this->completeRun();
     }
+
+    public function setLock(?LockInterface $lock = null): void
+    {
+        $reflection  = new \ReflectionClass($this);
+        $parentClass = $reflection->getParentClass();
+
+        if ($parentClass->hasProperty('lock')) {
```

Comment:
You can make it more readable by verifying the negative scenario first (the `lock` property does not exist), and changing its value later.
```suggestion
        if (!$parentClass->hasProperty('lock')) {
            throw new \RuntimeException("The 'lock' property does not exist in the parent class.");
        }
        
        $property = $parentClass->getProperty('lock');
        $property->setValue($this, $lock);
```

Reply:
Nice catch, thanks.